rma/models/rma.py

```diff
@@ -1233,86 +1331,27 @@ def create_replace(self, scheduled_date, warehouse, product, qty, uom):
                 )
                 + "\n"
             )
-        self.message_post(
-            body=body
-            or _(
-                "Replacement:<br/>"
-                'Product <a href="#" data-oe-model="product.product" '
-                'data-oe-id="%(id)d">%(name)s</a><br/>'
-                "Quantity %(qty)s %(uom)s<br/>"
-                "This replacement did not create a new move, but one of "
-                "the previously created moves was updated with this data."
-            )
-            % (
-                {
-                    "id": product.id,
-                    "name": product.display_name,
-                    "qty": qty,
-                    "uom": uom.name,
-                }
-            )
-        )
-        if self.state != "waiting_replacement":
-            self.state = "waiting_replacement"
-
-    def _action_launch_stock_rule(
-        self,
-        scheduled_date,
-        warehouse,
-        product,
-        qty,
-        uom,
-    ):
-        """Creates a delivery picking and launch stock rule. It is invoked by:
-        rma.create_replace
-        """
-        self.ensure_one()
-        if self.product_id.type not in ("consu", "product"):
-            return
-        if not self.procurement_group_id:
-            self.procurement_group_id = (
-                self.env["procurement.group"]
-                .create(
+        for rma in self:
```

Comment:
Don't convert everything to multi-record to get still more diff. Let it `self.ensure_one()` and call it one by one.